attestation-service/attestation-service/src/lib.rs

```diff
@@ -109,29 +71,55 @@ impl AttestationService {
             .map_err(|e| anyhow!("Cannot Set Policy: {:?}", e))
     }
 
+    fn accumulate_hash(materials: &[Vec<u8>]) -> Option<Vec<u8>> {
+        if materials.is_empty() {
+            return None;
+        }
+        let mut hasher = Sha384::new();
+        materials.iter().for_each(|m| hasher.update(m));
+        Some(hasher.finalize().to_vec())
+    }
+
```

Comment:
As part of the new TEE verifier API, I'm hoping we get to specify TEE specific hasher to make it symmetric to https://github.com/confidential-containers/guest-components/pull/369

Reply:
About the hash algorithm compatibility with like Intel TA/Amber, I'd like to add a parameter to the coco-as's api and decouple hash algorithm with verifier. However this can be done in another context and PR. WDYT?

Reply:
My preference is this PR since we are already doing refactoring.

Reply:
See https://github.com/confidential-containers/kbs/pull/216#discussion_r1399932779. As it took quite a while for the maintainers of different verifier plugin to review this. If the logic is not inside concrete plugin, it might be good for us to impl in another PR.

Reply:
Also, please double check the SGX verifier is good to you : )

Reply:
> See [#216 (comment)](https://github.com/confidential-containers/kbs/pull/216#discussion_r1399932779). As it took quite a while for the maintainers of different verifier plugin to review this. If the logic is not inside concrete plugin, it might be good for us to impl in another PR.

That's what I was saying: IMO, it's better to have the reportdata handling inside each attester/verifier and not try to make generic interpretation of it.

Reply:
The api inputs are straightforward the value of expected **hash**, not the **material data**.
```rust
pub trait Verifier {
  async fn evaluate(
          &self,
          evidence: &[u8],
          expected_report_data: Option<&[u8]>,
          expected_init_data_hash: Option<&[u8]>,
      ) -> Result<TeeEvidenceParsedClaim>;
}
```

How to calculate the **hash** from the **material data** should be defined outside the concrete `Verifier`.

Reply:
Try to think about this: if developers want to use `sgx` verifier, this decoupling design allows them to use any hash algorithms. About the handshake between AA and KBS, we can just choose one of them.